### PR TITLE
Change the calculation of the desired ray actors

### PR DESCRIPTION
**Pull request by revit13**
## Why are these changes needed?

Recent [PR](https://github.com/IBM/data-prep-kit/pull/616) reveals that the number of live ray actors does not reach the number of desired ray actors which causes the kfp workflow to eventually fail. This PR includes a more accurate calculation of the desired ray actors to fix that.

### CODE REVIEW

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/kfp_utils.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def default_compute_execution_params(
         cluster_gpu = w_options["replicas"] * w_options.get("gpu", 0.0)
         logger.info(f"Cluster available CPUs {cluster_cpu}, Memory {cluster_mem}, GPUs {cluster_gpu}")
         # compute number of actors
-        n_actors_cpu = int(cluster_cpu * 0.85 / a_options.get("num_cpus", 0.5))
+        n_actors_cpu = int((cluster_cpu - 1) * 0.7 / a_options.get("num_cpus", 0.5))
```
Comment (blublinsky) [inline kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/kfp_utils.py:141]: To make it symmetric, can we try:

 n_actors_cpu = int((cluster_cpu - 1) * 0.85 / a_options.get("num_cpus", 0.5))

File: transforms/universal/ededup/kfp_ray/src/ededup_compute_execution_params.py [hunk 2]
```diff
@@ -94,9 +95,9 @@ def ededup_compute_execution_params(
         )
         sys.exit(1)
     # Define number of workers
-    n_workers = int((0.85 * cluster_cpu - required_hash_cpu) / actor_cpu)
+    n_workers = int((0.85 * (cluster_cpu - 1) - required_hash_cpu) / actor_cpu)
     print(f"Number of workers - {n_workers}")
-    if n_workers < 2:
+    if n_workers < 0:
```
Comment (blublinsky) [inline transforms/universal/ededup/kfp_ray/src/ededup_compute_execution_params.py]: can we do:

if n_workers <= 0:

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/kfp_utils.py (hunk 1 above)
Comment (revit13) [reply kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/kfp_utils.py:141]: noop fails with this change...

File: transforms/universal/ededup/kfp_ray/src/ededup_compute_execution_params.py (hunk 2 above)
Comment (revit13) [reply transforms/universal/ededup/kfp_ray/src/ededup_compute_execution_params.py]: sure ededup runs ok with it

File: kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/kfp_utils.py (hunk 1 above)
Comment (blublinsky) [reply kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/kfp_utils.py:141]: lets put back

**Review by blublinsky — APPROVED** [review]
brilliant

**Pull request merged**